XMonad/Actions/CopyWindow.hs

```diff
@@ -88,6 +90,35 @@ import qualified XMonad.StackSet as W
 -- > main = do
 -- >    h <- spawnPipe "xmobar"
 -- >    xmonad def { logHook = sampleLogHook h }
+--
+-- Case 2: Using 'StatusBar'
+--
+-- First use a check function.
+--
+-- > checkCopy :: X (WorkspaceId -> String)
+-- > checkCopy = do
+-- >     copies <- wsContainingCopies
+-- >     let check ws | ws `elem` copies = pad . xmobarColor "red" "black" $ ws 
+-- >                  | otherwise = pad ws
+-- >     return check
+-- >
+-- > main = do
+-- >     let hs = statusBarPropTo' "_XMONAD_LOG_0" "xmobar -x 0 ~/.config/xmobar/xmobarrc0" myXmobarConfig
+-- >     xmonad $ withSB hs def
+--
+-- To integrate checkCopy into StatusBar, a modified version of statusBarPropTo is needed.
+--
+-- > statusBarPropTo' :: String -- ^ Property to write the string to
+-- >                 -> String -- ^ The command line to launch the status bar
+-- >                 -> PP   -- ^ The pretty printing options
+-- >                 -> StatusBarConfig
+-- > statusBarPropTo' prop cmd pp = def
+-- >     { sbLogHook     = xmonadPropLog' prop =<< dynamicLogString . (\w -> pp { ppHidden = w }) =<< checkCopy
+-- >     , sbStartupHook = spawnStatusBarAndRemember cmd
+-- >     , sbCleanupHook = cleanupStatusBars
+-- >     }
```

Comment:
The whole point of the abstractions is to not fiddle with low-level things anymore. All the `statusBar*` functions accept an `X PP`, so it's best to write a function that takes a `PP` and makes it aware of copies. Something like `copiesPP :: (WorkspaceId -> String) -> PP -> X PP`, and then apply it like so `statusBarProp "xmobar" (copiesPP myContainsCopies def)`

I also think that the examples should be minimal: No need to use `statusBarPropTo` with `xmobar -x 0`. A better way is to prefer `statusBarProp "xmobar"` and make the readers read the documentation to see what other options are possible

Reply:
I have upated the example, thanks.

Reply:
I think you missed my point, why not just add `copiesPP` as a library function?